workspaces/src/result.rs

```diff
@@ -100,15 +103,34 @@ impl CallExecutionDetails {
         }
     }
 
-    /// Convert the execution details into a Result if its status is not a successful one.
-    /// Useful for checking if the call was successful and forwarding the error upwards.
-    fn try_into_result(self) -> Result<Self> {
+    /// Checks whether the transaction was executed successfully. This will return
+    /// a `Ok(())` if the transaction was successful, and a `Err(Error)` if it was not.
+    ///
+    /// This acts a convenience function to allow us to forward errors when we do not
+    /// care about the details of the transaction at all.
+    pub fn executed(&self) -> Result<()> {
         self.try_into_success_value()?;
-        Ok(self)
+        Ok(())
```

Comment:
It's the same error type, isn't it?

can just do (can't add suggestion on these lines):
```rust
self.try_into_success_value()
```

Reply:
This is actually returning `Result<&str>` so wouldn't be able to forward it